test/jdk/java/lang/invoke/MethodHandleProxies/ProxiesInterfaceTest.java

```diff
@@ -0,0 +1,259 @@
+                "multiple abstract method names");
+        assertThrows(IllegalArgumentException.class, () -> asInterfaceInstance(NoAbstractMethods.class, mh),
+                "no abstract method");
+        assertThrows(IllegalArgumentException.class, () -> asInterfaceInstance(Sealed.class, mh),
+                "sealed interface");
+    }
+
+    /**
+     * Tests that non-sealed interfaces can be implemented.
+     */
+    @Test
+    public void testNonSealed() {
+        MethodHandle target = MethodHandles.constant(String.class, "Non-Sealed");
+        NonSealed proxy = asInterfaceInstance(NonSealed.class, target);
+        assertEquals(proxy.m(), "Non-Sealed");
+    }
+
+    /**
+     * Tests that Proxy correctly proxies potential bridge abstract methods.
+     */
+    @Test
+    public void testMultiSameName() throws Throwable {
+        var baseAndChild = loadBaseAndChild();
+        var baseClass = baseAndChild.get(0);
+        var childClass = baseAndChild.get(1);
+        checkMethods(childClass.getMethods());
+        checkMethods(childClass.getDeclaredMethods());
+
+        var lookup = MethodHandles.lookup();
+        var baseValueMh = lookup.findVirtual(baseClass, "value", genericMethodType(0))
+                .asType(genericMethodType(1));
+        var childIntegerValueMh = lookup.findVirtual(childClass, "value", methodType(Integer.class))
+                .asType(methodType(Integer.class, Object.class));
+        var childIntValueMh = lookup.findVirtual(childClass, "value", methodType(int.class))
+                .asType(methodType(int.class, Object.class));
+
+        Object child = asInterfaceInstance(childClass, MethodHandles.constant(Integer.class, 7));
+
+        assertEquals(7, (Object) baseValueMh.invokeExact(child));
+        assertEquals(7, (Integer) childIntegerValueMh.invokeExact(child));
+        assertEquals(7, (int) childIntValueMh.invokeExact(child));
+    }
+
+    /**
+     * Tests that default methods can be used.
+     */
+    @Test
+    public void testDefaultMethods() {
+        MethodHandle target = MethodHandles.constant(String.class, "F");
+        C proxy = asInterfaceInstance(C.class, target);
+
+        assertEquals(proxy.f(), "F");
+        assertEquals(proxy.a(), "A");
+        assertEquals(proxy.b(), "B");
+        assertEquals(proxy.c(), "C");
+        assertEquals(proxy.concat(), "ABC");
+    }
+
+    /**
+     * Tests that correct implementation of default methods are called,
+     * and correct abstract methods are implemented.
+     */
+    @Test
+    public void testOverrides() {
+        MethodHandle target = MethodHandles.constant(String.class, "concat");
+        D proxy = asInterfaceInstance(D.class, target);
+
+        assertEquals(proxy.a(), "OA");
+        assertEquals(proxy.b(), "OB");
+        assertEquals(proxy.c(), "OC");
+        assertEquals(proxy.f(), "OF");
+        assertEquals(proxy.concat(), "concat");
+    }
+
+    //<editor-fold desc="Infrastructure">
```

Comment:
this comment can be deleted.

Reply:
It was a leftover from when I tried to organize the tests with IDE editor folds. No longer necessary indeed.